types/jasmine/index.d.ts

```diff
@@ -554,6 +554,7 @@ declare namespace jasmine {
         toThrowMatching(predicate: (thrown: any) => boolean): boolean;
         toBeNegativeInfinity(expectationFailOutput?: any): boolean;
         toBePositiveInfinity(expectationFailOutput?: any): boolean;
+        toBeInstanceOf(expected: any): boolean;
```

Comment:
Could you please verify whether we couldn't use more specific type here? As far as I can see, only prototype must be allowed. Probably, we could use `jasmine.Constructor` instead.

Also please add the tests to verify that it compiles for prototypes and fails for arbitrary values. Ensure to keep both "pre 3.1" and "3.1+" types in sync, so that they match line by line.